blockdag/process.go

```diff
@@ -164,19 +164,26 @@ func (dag *BlockDAG) ProcessBlock(block *util.Block, flags BehaviorFlags) (isOrp
 		}
 
 		if delay != 0 {
+			dag.AddDelayedBlock(block, delay)
 			return false, delay, err
 		}
 	}
 
-	// Handle orphan blocks.
-	allParentsExist := true
+	var missingParents []*daghash.Hash
 	for _, parentHash := range block.MsgBlock().Header.ParentHashes {
 		if !dag.BlockExists(parentHash) {
-			allParentsExist = false
+			missingParents = append(missingParents, parentHash)
 		}
 	}
 
-	if !allParentsExist {
+	// Handle the case of a block with a valid timestamp(non-delayed) which points to a delayed block.
+	delay = dag.maxDelayOfParents(missingParents)
```

Comment:
What if block has both delayed parents, and genuinely missing parents orphan-style?

At the very least - the missing parents should be requested.

Also, I don't think we are re-validating all parents are present once delay is over.

Reply:
It seems like you forgot to add the block with delayed parent to the delayed pool